platforms/nuttx/src/canbootloader/uavcan/main.c

```diff
@@ -1185,6 +1185,18 @@ __EXPORT int main(int argc, char *argv[])
 
 	} else {
 
+		// JAKE: It is a regular boot, so we should just jump straight to the app because
+		// the battery is always powered on before the flight controller.
+		// Except in dual battery systems... In which case.. what happens?
+
+		// If app is already valid, jump straight to app
+		if (bootloader.app_valid) {
```

Comment:
I think this will break the "Breakproof guaranty" of the UAVCAN bootloader. A CRC valid app does not guarantee the app can run and be involved in replacement. I.E. A bug in the app  

This could be done on watch dogged system using state to to insure the app ran and can reboot with a NODE ID.

Reply:
I see your point. The smart battery use case is unique in that it turns the FC on, therefore it's not possible to get a Node ID until the FC has booted. What other methods can we use to determine that the app is working normally? Perhaps we could use a parameter CANNODE_FASTBOOT that comes with a disclaimer that enabling fast boot on an untested firmware breaks the breakproof guarantee.

Reply:
There is a jumper option. We could use that to unbrick. But this should be a CONFIG_ set for the BL build that defaults to how it works now.

Reply:
I'd strongly prefer to make this behavior the norm if we can.

Any concerns with requiring the watch dog for the "guarantee"?

Reply:
Nope  but we will need a "Stay in bootloader" (RTC signature) and this adds a dependency for nonvol or a shared memory communication.